docs/docs/reference/contextual/right-associative-extension-methods.md

```diff
@@ -0,0 +1,45 @@
+---
+layout: doc-page
+title: "Right-Associative Extension Methods: Details"
+---
+
+The most general form of leading parameters of an extension method is as follows:
+
+  - A possibly empty list of using clauses `leadingUsing`
+  - A single parameter `extensionParam`
+  - A possibly empty list of using clauses `trailingUsing`
+
+This is then followed by `def`, the method name, and possibly further parameters
+`otherParams`. An example is:
+
+```scala
+  extension (using a: A, b: B)(using c: C)    // <-- leadingUsing
+            (x: X)                            // <-- extensionParam
+            (using d: D)                      // <-- trailingUsing
+    def +:: (y: Y)(using e: E)(z: Z)          // <-- otherParams
+```
+An extension method is treated as a right associative operator if
+it has a name ending in `:` and is immediately followed by a
+single parameter. In the example above, that parameter is `(y: Y)`.
+
+The Scala compiler pre-processes a right-associative infix operation such as `x +: xs`
+to `xs.+:(x)` if `x` and `xs` are pure expressions or `x` is a call-by-name parameter and to `val y = x; xs.+:(y)` otherwise. This is necessary since a regular right-associative infix method
```

Comment:
Sorry for commenting the same place twice but it was only now that I spotted it:
If `xs` is impure  but `x` is pure there should be no assignment, right? In general (im)purity of `xs` doesn't seem to play any role here. So it probably should be
```
if `x` is a pure expressions or a call-by-name parameter
```

Reply:
Yes, we can make the change. I leave that to you since I am working on something else now.

Reply:
It looks like I can't push to this branch as I don't have permissions to dotty-staging. I didn't even notice that before as I created PRs from my own fork.

Reply:
I see you were not added yet. I just invited you with push rights to dotty-staging.

Reply:
Thanks. Changes pushed